### PR TITLE
docs: Update console log formatter tracing docs

### PR DESCRIPTION
**Pull request by jskeet**

### CODE REVIEW

**Comment by amanda-tarafa** [comment]
> This is only a first pass - I need to make sure I understand a few more things first.

But it does look good.

**Comment by jskeet** [comment]
Second commit is ready - PTAL.

**Comment by jskeet** [comment]
(I'll rebase into one eventually, of course.)

File: apis/Google.Cloud.Logging.Console/docs/index.md [hunk 1]
```diff
@@ -33,26 +33,31 @@ builder.Logging.AddGoogleCloudConsole(options => options.IncludeScopes = true);
...
-
-The code to enable log trace correlation is shown below:
+The console formatter supports [log correlation](https://cloud.google.com/logging/docs/view/correlate-logs) and
+[Google Cloud Trace log integration](https://cloud.google.com/trace/docs/trace-log-integration) via trace IDs.
+This requires a number of conditions to be met:
+
+- Your application needs to be exporting log entries to Google Cloud Logging via an agent that reads from stdout.
+  This happens automatically when running on Google Cloud, and is presumably the reason you're using this
+  console formatter.
+- Your application needs to be exporting traces to Google Cloud Trace. This could be performed automatically
```
Comment (jskeet) [inline apis/Google.Cloud.Logging.Console/docs/index.md:43]: This probably *isn't* actually required for just log correlation, but we'll validate that (if we have time) after merging and releasing.

File: apis/Google.Cloud.Logging.Console/docs/index.md [hunk 2]
```diff
@@ -33,18 +33,23 @@ builder.Logging.AddGoogleCloudConsole(options => options.IncludeScopes = true);
...
+The console formatter supports [log correlation](https://cloud.google.com/logging/docs/view/correlate-logs) and
+[Google Cloud Trace log integration](https://cloud.google.com/trace/docs/trace-log-integration) via trace IDs.
+This requires a number of conditions to be met:
 
 - Your application needs to be exporting log entries to Google Cloud Logging via an agent that reads from stdout.
   This happens automatically when running on Google Cloud, and is presumably the reason you're using this
   console formatter.
 - Your application needs to be exporting traces to Google Cloud Trace. This could be performed automatically
   by the runtime, or via explicit code or instrumentation. Google Cloud runtimes export some traces automatically.
+  (For example, Cloud Run will [export traces for all received requests](https://cloud.google.com/run/docs/trace)
```
Comment (amanda-tarafa) [inline apis/Google.Cloud.Logging.Console/docs/index.md:45]: It's not for all received requests, it does sampling. Just removing the "all" should be fine.

**Review by amanda-tarafa — APPROVED** [review]
Tiny nit, otherwise LGTM

File: apis/Google.Cloud.Logging.Console/docs/index.md (hunk 2 above)
Comment (jskeet) [reply apis/Google.Cloud.Logging.Console/docs/index.md:45]: Great, will do, thanks.

**Pull request merged**